animations/tween-animations.md

```diff
@@ -0,0 +1,727 @@
+  @override
+  _LogoAppState createState() => new _LogoAppState();
+}
+
+class _LogoAppState extends State<LogoApp> with TickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  initState() {
+    super.initState();
+    controller = new AnimationController(
+        duration: const Duration(milliseconds: 2000), vsync: this);
+    animation = new Tween(begin: 0.0, end: 300.0).animate(controller);
+    controller.forward();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new AnimatedLogo(animation: animation);
+  }
+
+  @override
+  dispose() {
+    super.dispose();
+    controller.dispose();
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+{% endprettify %}
+
+LogoApp passes the Animation object to the base class and uses
+`animation.value` to set the height and width of the container, so
+it works exactly the same as before.
+
+You can find the source for this example,
+[animate2,](https://raw.githubusercontent.com/InMatrix/animation_tutorial/master/lib/animate2.dart)
+on GitHub.
+
+<a name="monitoring"></a>
+## Monitoring the progress of the animation
+
+It’s often helpful to know when an animation changes state,
+such as finishing, moving forward, or reversing.
+You can get notifications for this with `addStatusListener()`.
+The following code modifies the previous example so that
+it listens for a state change and prints an update.
+The highlighted line shows the change:
+
+<!-- skip -->
+{% prettify dart %}
+class _LogoAppState extends State<LogoApp> with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  initState() {
+    super.initState();
+    controller = new AnimationController(
+        duration: const Duration(milliseconds: 2000), vsync: this);
+    animation = new Tween(begin: 0.0, end: 300.0).animate(controller)
+      [[highlight]]..addStatusListener((state) => print("$state"));[[/highlight]]
+    controller.forward();
+  }
+  //...
+}
+{% endprettify %}
+
+Running this code produces lines like the following:
+
+<!-- skip -->
+{% prettify sh %}
+AnimationStatus.forward
+AnimationStatus.completed
+{% endprettify %}
+
+Next, use `addStatusListener()` to reverse the animation at the
+beginning or the end. This creates a "breathing" effect:
+
+<!-- skip -->
+{% prettify dart %}
+class _LogoAppState extends State<LogoApp> with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  initState() {
+    super.initState();
+    controller = new AnimationController(
+        duration: const Duration(milliseconds: 2000), vsync: this);
+    animation = new Tween(begin: 0.0, end: 300.0).animate(controller);
+
+    animation.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        controller.forward();
+      }
+    });
+    controller.forward();
+  }
```

Comment:
you can also get this same effect with an appropriately configured sawtooth tween and "repeat" on the animation controller.